CMake tries to read a list field from SDKSettings.plist, but the output of
facebook/xcbuild PlistBuddy is incompatible with Apple's.

Simply set the supported architectures to the one target architecture we're
building for.

--- a/cmake/modules/SwiftConfigureSDK.cmake
+++ b/cmake/modules/SwiftConfigureSDK.cmake
@@ -189,7 +189,7 @@ macro(configure_sdk_darwin
   endif()
 
   # Remove any architectures not supported by the SDK.
-  remove_sdk_unsupported_archs(${name} ${xcrun_name} ${SWIFT_SDK_${prefix}_PATH} SWIFT_SDK_${prefix}_ARCHITECTURES)
+  set(SWIFT_SDK_${prefix}_ARCHITECTURES "@swiftArch@")
 
   list_intersect(
     "${SWIFT_DARWIN_MODULE_ARCHS}"            # lhs
